Inc/HALALMock/Models/PinModel/Pin.hpp

```diff
@@ -78,6 +78,65 @@ enum PinState{
 	ON
 };
 
+enum PinType {
+	DigitalOutput,
+	DigitalInput,
+	PWM,
+	ADC
+	// TODO: Add more types
+};
+
+struct PinModel {
```

Comment:
```suggestion
struct EmulatedPin {
```
I really don't know how to name this, I don't mind `PinModel`, but at the same time I think we could come up with a better name. Think about it but if you don't find a better name just leave it as is. I think the name I propose in the comment gives a clear impression that is related to the simulator.